extensions/objects/templates/ImageWithRegionsEditor.js

```diff
@@ -233,6 +290,14 @@ oppia.directive('imageWithRegionsEditor', [
           }
           $scope.userIsCurrentlyDrawing = false;
           $scope.userIsCurrentlyDragging = false;
+          $scope.userIsCurrentlyResizing = false;
+          if ($scope.outOfRegion) {
+            $scope.xDirection = 0;
+            $scope.yDirection = 0;
+          }
```

Comment:
The reason for using this condition and the variable $scope.outOfRegion is whenever a user starts resizing from say North direction then the pointer can be placed anywhere between y and y+10(resizeRegionBorder) and the pointer is n-resize. Now while resizing if the direction is reversed by going in opposite direction then the pointer direction is reversed to south and the pointer is s-resize. Now initially the pointer was below the top y-coordinate, when resize began  in the opposite direction(i,e. south and on the same mouse press) the pointer goes below the lower y coordinate by the same initial margin. Hence in that case the pointer need to be reset to default.
If these are always set to 0 on mouseUp then if I want to resize from the same position where I left then I won't be able to do so as the xDirection and yDirection are not set.